toolkit/tools/specreader/specreader.go

```diff
@@ -82,13 +84,55 @@ func main() {
 	toolchainRPMs, err := schedulerutils.ReadReservedFilesList(*toolchainManifest)
 	logger.PanicOnError(err, "Unable to read toolchain manifest file '%s': %s", *toolchainManifest, err)
 
-	err = parseSPECsWrapper(*buildDir, *specsDir, *rpmsDir, *srpmsDir, *existingToolchainRpmDir, *distTag, *output, *workerTar, toolchainRPMs, *workers, *runCheck)
+	// A parse list may be provided, if so only parse this subset.
+	// If none is provided, parse all specs.
+	parseList, err := parseParseListFile(*parseListFile)
 	logger.PanicOnError(err)
+
+	// Convert specsDir to an absolute path
+	specsAbsDir, err := filepath.Abs(*specsDir)
+	logger.PanicOnError(err, "Unable to get absolute path for specs directory '%s': %s", *specsDir, err)
+
+	err = parseSPECsWrapper(*buildDir, specsAbsDir, *rpmsDir, *srpmsDir, *existingToolchainRpmDir, *distTag, *output, *workerTar, parseList, toolchainRPMs, *workers, *runCheck)
+	logger.PanicOnError(err)
+}
+
+// parsePackListFile will parse a list of packages to pack if one is specified.
+// Duplicate list entries in the file will be removed.
+func parseParseListFile(parseListFile string) (parseList map[string]bool, err error) {
```

Comment:
parseParseListFile sounds weird to me. Maybe parseSpecListFile? This will naturally flow from above renaming I suggested
Is this a generic parser we're making, which will be used for other purposes besides just parsing the packages here? Then, we could rename to parseListFile?

Reply:
Refactored to a new package now